lookout/style/format/analyzer.py

```diff
@@ -296,37 +321,27 @@ def _generate_file_comments(
         assert len(y) == len(rule_winners)
         code_generator = CodeGenerator(fe, skip_errors=True)
         new_vnodes = code_generator.apply_predicted_y(vnodes, vnodes_y, y_pred)
-        code_lines = file.content.decode("utf-8", "replace").splitlines()
         for line_number, line in self._group_line_nodes(
                 y, y_pred, vnodes_y, new_vnodes, rule_winners):
             line_ys, line_ys_pred, line_vnodes_y, new_line_vnodes, line_winners = line
             new_code_line = code_generator.generate(
                 new_line_vnodes, "local").lstrip("\n").splitlines()[0]
-            fix_rules = []
-            fix_vnodes = []
-            fix_ys = []
-            for line_yi, line_y_predi, line_vnode_y, line_winner in zip(
-                    line_ys, line_ys_pred, line_vnodes_y, line_winners):
-                if line_yi == line_y_predi:
-                    continue
-                fix_rules.append(line_winner)
-                fix_vnodes.append(line_vnode_y)
-                fix_ys.append(line_y_predi)
             confidence = self._get_comment_confidence(line_ys, line_ys_pred, line_winners, rules)
-            text = self.render_comment_text(
-                language=language,            # programming language of the code
-                line_number=line_number,      # line number for the comment
-                code_lines=code_lines,        # original file code lines
-                new_code_line=new_code_line,  # code line suggested by our model
-                winners=fix_rules,            # winner rule indices
-                vnodes=fix_vnodes,            # corrected VirtualNode-s
-                fixed_labels=fix_ys,          # predicted labels for those nodes
-                confidence=confidence,        # overall confidence in the prediction, 0-100
-                feature_extractor=fe,         # FeatureExtractor involved.
+            vnodes_changed = [vnode for vnode in new_line_vnodes if
+                              hasattr(vnode, "y_old") and vnode.y_old != vnode.y]
+            yield FixData(
+                error="",                      # success
+                language=language,             # programming language of the code
+                line_number=line_number,       # line number for the comment
+                head_file=file,                # file from head revision
+                base_file=base_file,           # file from base revision
+                suggested_code=new_code_line,  # code line suggested by our model
+                all_vnodes=new_line_vnodes,    # VirtualNode-s which construct the line
```

Comment:
all_vnodes correspond to the whole file
vnodes correspond to the whole line

so

```python
all_vnodes=new_vnodes
vnodes=new_line_vnodes
```